src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -698,22 +693,20 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 	}
 
 	private getModels(): ILanguageModelChatMetadataAndIdentifier[] {
-		const models = this.languageModelsService.getLanguageModelIds()
-			.map(modelId => ({ identifier: modelId, metadata: this.languageModelsService.lookupLanguageModel(modelId)! }))
-			.filter(entry => entry.metadata?.isUserSelectable && this.modelSupportedForDefaultAgent(entry));
+		const cachedModels = this.storageService.getObject<ILanguageModelChatMetadataAndIdentifier[]>('chat.cachedLanguageModels', StorageScope.APPLICATION, []);
```

Comment:
Why cache these in the view, can the ILanguageModelsService handle the caching?

Reply:
Caching in the view state makes more sense as a lot of things rely on the language model service as the actual data source for when things are ready. Such as the chat setup / proxy. Effectively what caching here allows us to do is have up to date UI state while having all the proper event handlers + gets on the language model service be reserved for "real" models